zfit/core/parameter.py

```diff
@@ -619,6 +622,17 @@ def __init__(self, params, value_fn, name="BaseComposedParameter", **kwargs):
         super().__init__(name=name, params=params, **kwargs)
         if not callable(value_fn):
             raise TypeError("`value_fn` is not callable.")
+        n_params = len(signature(value_fn).parameters)
+        # TODO(0.6): remove, legacy?
+        if n_params == 0:
```

Comment:
That's the legacy hack, maybe just have a look at these few lines, the rest is fine

Reply:
Maybe give a proper deprecation warning?

Reply:
And you are not testing for what you are claiming to test in the warning.

Reply:
Yes, true. I've added the real test and changed to a depr warning